src/utils.js

```diff
@@ -25,28 +25,96 @@ export const addTile = (layer, tile, x, y) => {
     return sprite;
 }
 
-class Tile {
-    constructor(texture, tileX, tileY) {
-        this.frame = texture.frame(new Point(tileX * TILE_SIZE, tileY * TILE_SIZE), new Point(TILE_SIZE, TILE_SIZE));
-    }
-}
 const textures = [];
-export const getTile = (src, sx, sy) =>
-    new Promise(resolve => {
+export async function loadTexture(src) {
+    return await new Promise(resolve => {
         if (textures[src]) {
-            const tile = new Tile(textures[src], sx, sy);
-            resolve(tile);
+            resolve(textures[src]);
             return;
         } else {
             const image = new Image;
             image.onerror = image.onload = () => {
                 const texture = scene.texture(image, 1);
                 textures[src] = texture;
-                const tile = new Tile(texture, sx, sy);
-                resolve(tile);
+                resolve(texture);
             }
             image.src = src;
         }
     });
+}
+
+const atlasTexture = await loadTexture("src/atlas.png");
+
+export class TileResource {
+    constructor(tileX, tileY) {
+        this.frame = atlasTexture.frame(new Point(tileX * TILE_SIZE, tileY * TILE_SIZE), new Point(TILE_SIZE, TILE_SIZE));
+    }
+}
+
+export class AnimResource {
+    constructor(origin, size, numFrames, options) {
+        if (options == null)
+            options = {};
```

Comment:
```suggestion
    constructor(origin, size, numFrames, options = {}) {
```

sinon on peut garder null mais il faut utiliser 
```
option?.xxx || defaultValue;
```